code/client/munkilib/Repo.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/python
+# encoding: utf-8
+#
+# Copyright 2016 Centrify Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Repo
+Created by Centrify Corporation 2016-06-02.
+Interface for accessing a repo.
+"""
+
+import re
+import sys
+import imp
+
+def Open(path, url, plugin):
+    #looks for plugin in /usr/local/munki/munkilib/plugins (installation of munki)
+    if plugin == None or plugin == "":
+        #default is FileRepo if no plugin is specified in configuration or options.
+        module = imp.load_source('FileRepo', '/usr/local/munki/munkilib/FileRepo.py')
+        import_class = getattr(module, "FileRepo")
+        parent = import_class
+    else:
+        module = imp.load_source(plugin, '/usr/local/munki/munkilib/plugins/' + plugin + ".py")
+        import_class = getattr(module, plugin)
```

Comment:
Again I dislike all the hard-coding of install locations here -- let's get this working with relative locations. Perhaps also the FileRepo module should be in the plugins directory like any other repo plugin